lib/faraday_middleware/gzip.rb

```diff
@@ -48,7 +48,17 @@ def uncompress_gzip(body)
     end
 
     def inflate(body)
+      # Inflate as a DEFLATE (RFC 1950+RFC 1951) stream
       Zlib::Inflate.inflate(body)
+    rescue Zlib::DataError
```

Comment:
I understand. No, I think your implementation right now with the rescue is the sanest and simplest implementation for now. :+1: